speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/ResearcherMenu.cpp

```diff
@@ -161,6 +175,51 @@ static void SetUserId(void*)
     GfuiEditboxSetString(s_scrHandle, m_userIdControl, m_userId);
 }
 
+/// @brief Saves the settings into the ResearcherMenu.xml file
+static void SaveSettingsToDisk() 
+{
+    // Copies xml to documents folder and then ospens file parameter
+    std::string dstStr("config/ResearcherMenu.xml");
+    char dst[512];
+    sprintf(dst, "%s%s", GfLocalDir(), dstStr.c_str());
+    void* readParam = GfParmReadFile(dst, GFPARM_RMODE_REREAD | GFPARM_RMODE_CREAT);
+
+    // Save task settings to xml file
+    char val[32];
+    sprintf(val, "%d", m_task);
+    GfParmSetStr(readParam, RESEARCHMENU_TASKS, GFMNU_ATTR_SELECTED, val);
+
+    // Save indicator settings to xml file
+    const char* audioSetting = m_indicators.Audio ? GFMNU_VAL_YES : GFMNU_VAL_NO;
+    const char* iconSetting  = m_indicators.Icon  ? GFMNU_VAL_YES : GFMNU_VAL_NO;
+    const char* textSetting  = m_indicators.Text  ? GFMNU_VAL_YES : GFMNU_VAL_NO;
+    GfParmSetStr(readParam, RESEARCHMENU_INDICATOR_AUDITORY, GFMNU_ATTR_CHECKED, audioSetting);
+    GfParmSetStr(readParam, RESEARCHMENU_INDICATOR_VISUAL,   GFMNU_ATTR_CHECKED, iconSetting);
+    GfParmSetStr(readParam, RESEARCHMENU_INDICATOR_TEXT,     GFMNU_ATTR_CHECKED, textSetting);
+
+    // Save intervention type settings to xml file
+    sprintf(val, "%d", m_interventionType);
+    GfParmSetStr(readParam, RESEARCHMENU_INTERVENTIONTYPE, GFMNU_ATTR_SELECTED, val);
+
+    // Save participant control settings to xml file
+    const char* controlGas         = m_pControl.ControlGas                ? GFMNU_VAL_YES : GFMNU_VAL_NO;
+    const char* interventionToggle = m_pControl.ControlInterventionToggle ? GFMNU_VAL_YES : GFMNU_VAL_NO;
+    const char* controlSteering    = m_pControl.ControlSteering           ? GFMNU_VAL_YES : GFMNU_VAL_NO;
+    const char* forceFeedback      = m_pControl.ForceFeedback             ? GFMNU_VAL_YES : GFMNU_VAL_NO;
+    GfParmSetStr(readParam, RESEARCHMENU_CONTROL_GAS,                 GFMNU_ATTR_CHECKED, controlGas);
+    GfParmSetStr(readParam, RESEARCHMENU_CONTROL_INTERVENTION_TOGGLE, GFMNU_ATTR_CHECKED, interventionToggle);
+    GfParmSetStr(readParam, RESEARCHMENU_CONTROL_STEERING,            GFMNU_ATTR_CHECKED, controlSteering);
+    GfParmSetStr(readParam, RESEARCHMENU_FORCE_FEEDBACK,              GFMNU_ATTR_CHECKED, forceFeedback);
```

Comment:
It would be nice if you could add a function `GfParmSetBoolean` function next to the `GfParmSetStr` functions, that abstracts away the ternary condition that you are using everywhere now. 

Then you could just call `GfParmSetBoolean(param, path, name, boolean)` and this function will interally convert the boolean to YES/NO string values and use GfParmSetStr to actually set it.